src/systems/model_photo_shoot/ModelPhotoShoot.cc

```diff
@@ -0,0 +1,283 @@
+      {
+        // Compute the translation we have to apply to the cameras to
+        // center the model in the image.
+        ignition::math::AxisAlignedBox bbox = modelVisual->LocalBoundingBox();
+        double scaling = 1.0 / bbox.Size().Max();
+        ignition::math::Vector3d bboxCenter = bbox.Center();
+        ignition::math::Vector3d translation =
+            bboxCenter + this->modelPose3D.Pos();
+        if (this->savingFile.is_open()) {
+          this->savingFile << "Translation: " << translation << std::endl;
+          this->savingFile << "Scaling: " << scaling << std::endl;
+        }
+
+        ignition::math::Pose3d pose;
+        // Perspective view
+        pose.Pos().Set(1.6 / scaling + translation.X(),
+                       -1.6 / scaling + translation.Y(),
+                       1.2 / scaling + translation.Z()+0.078);
+        pose.Rot().Euler(0, IGN_DTOR(30), IGN_DTOR(-225));
+        SavePicture(camera, pose, "1.png");
+
+        // Top view
+        pose.Pos().Set(0 + translation.X(),
+                       0 + translation.Y(),
+                       2.2 / scaling + translation.Z()+0.078);
+        pose.Rot().Euler(0, IGN_DTOR(90), 0);
+        SavePicture(camera, pose, "2.png");
+
+        // Front view
+        pose.Pos().Set(2.2 / scaling + translation.X(),
+                       0 + translation.Y(),
+                       0 + translation.Z()+0.078);
+        pose.Rot().Euler(0, 0, IGN_DTOR(-180));
+        SavePicture(camera, pose, "3.png");
+
+        // Side view
+        pose.Pos().Set(0 + translation.X(),
+                       2.2 / scaling + translation.Y(),
+                       0 + translation.Z()+0.078);
+        pose.Rot().Euler(0, 0, IGN_DTOR(-90));
+        SavePicture(camera, pose, "4.png");
+
+        // Back view
+        pose.Pos().Set(-2.2 / scaling + translation.X(),
+                       0 + translation.Y(),
+                       0 + translation.Z()+0.078);
+        pose.Rot().Euler(0, 0, 0);
+        SavePicture(camera, pose, "5.png");
+
+        this->takePicture = false;
+      }
+    }
+  }
+}
+
+//////////////////////////////////////////////////
+void ModelPhotoShootPrivate::SavePicture(
+                                  const ignition::rendering::CameraPtr _camera,
+                                  const ignition::math::Pose3d _pose,
+                                  const std::string _fileName) const
+{
+  unsigned int width = _camera->ImageWidth();
+  unsigned int height = _camera->ImageHeight();
+  ignition::common::Image image;
+
+  _camera->SetWorldPose(_pose);
+  auto cameraImage = _camera->CreateImage();
+  _camera->Capture(cameraImage);
+  auto formatStr =
+      ignition::rendering::PixelUtil::Name(_camera->ImageFormat());
+  auto format = ignition::common::Image::ConvertPixelFormat(formatStr);
+  image.SetFromData(cameraImage.Data<unsigned char>(), width, height, format);
+  std::string fullName = _fileName;
```

Comment:
Any reason for the copy of `_fileName` into `fullName` here?

Reply:
Nope, previously I was adding the extension at this point but since it's not done anymore we don't need this.